File: JavaScript/7js 2/nextHappyYear.js
/*
|--------------------------------------------------------------------------
| See You Next Happy Year
|--------------------------------------------------------------------------
|
*/

function nextHappyYear(year) {
   while (new Set([...(++year + "")]).size < 4);
   return year;
}


function nextHappyYear(year) {
   return new Set(`${++year}`).size < 4 ? nextHappyYear(year) : year;
}


function nextHappyYear(year) {
   do year += 1;
   while (/([0-9]).*?\1/.test(year));
   return year;
}


function nextHappyYear(year) {
   while (new Set(String(++year).split("")).size < 4);
   return year;
}


console.log(nextHappyYear(1001), 1023);
console.log(nextHappyYear(1123), 1203);
console.log(nextHappyYear(2001), 2013);
console.log(nextHappyYear(2334), 2340);

File: Python/7py 2/next_happy_year.py
"""
|--------------------------------------------------------------------------
| See You Next Happy Year
|--------------------------------------------------------------------------
|
"""

def next_happy_year(year):
   year += 1
   while len(set(str(year))) != 4:
      year += 1
   return year


def next_happy_year(year):
   return year+1 if len(set(str(year+1))) == 4 else next_happy_year(year+1)


def next_happy_year(year):
   while True:
      year+=1
      if len(str(year)) == len(set(str(year))):
         return year


print(next_happy_year(1001),1023)
print(next_happy_year(1123),1203)
print(next_happy_year(2001),2013)
print(next_happy_year(2334),2340)